src/redux/restReducers/restReducer.ts

```diff
@@ -11,8 +11,28 @@ export interface ActionTypes {
 
 export interface RestReducer<T> {
     status: STATUS;
+}
+
+export interface Loaded<T> extends RestReducer<T> {
+    status:
+        | STATUS.SUCCESS
+        | STATUS.RELOADING;
     data: T;
-    error?: String;
+}
+
+export interface NoData<T> extends RestReducer<T> {
+    status:
+        | STATUS.NOT_STARTED
+        | STATUS.LOADING;
+}
+
+export interface Failed<T> extends RestReducer<T> {
+    status: STATUS.FAILED;
+    error: string;
+}
+
+export function isLoaded<T>(restReducer: RestReducer<T>): restReducer is Loaded<T> {
```

Comment:
Hva vil det si at den returnerer `restReducer is Loaded<T>`? Og kan den likegodt returnere en boolean eller har denne notasjonen noen ekstra verdi?

Reply:
Det er keywordet for å lage en typeguard. Penere enn STATUS.SUCCESS ==.... og da er objektet håndtert som den typen resten av grenen så man kan bruke f.eks:
isLoaded(props.utbetalingReducer) && visCheckbokser(props.utbetalingReducer**.data**)

Reply:
aha, nice!